reflect.libsonnet

```
@@ -0,0 +1,47 @@
+local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
+
+{
+  '#': d.pkg(
+    name='reflect',
+    url='github.com/jsonnet-libs/xtd/reflect.libsonnet',
+    help='`reflect` implements helper functions for processing json data.',
+  ),
+
+  '#prune':: d.fn(
+    '`prune` works the same as `std.prune` but with the options to disable recursion and only pruning `null` values.',
+    [
+      d.arg('a', d.T.any),
+      d.arg('recurse', d.T.bool, default=true),
+      d.arg('only_null', d.T.bool, default=false),
+    ]
+  ),
+  prune(a, recurse=true, only_null=false)::
+    local isContent(b) =
+      if b == null
+      then false
+      else if only_null
+      then true
+      else if std.isArray(b)
+      then std.length(b) > 0
+      else if std.isObject(b)
+      then std.length(b) > 0
+      else true;
+    local doRecursion(x) =
+      if recurse
+      then self.prune(x, only_null=only_null)
```

nit: I think it's more common to see that the recursive call uses the exact same arguments as the initial call because it explicitly demonstrates that the recursive call is gonna continue the same behavior.

```suggestion
      then self.prune(x, recurse, only_null)
```